src/EspMQTTClient.cpp

```diff
@@ -692,6 +692,14 @@ bool EspMQTTClient::mqttTopicMatch(const String &topic1, const String &topic2)
       if(topic2.substring(t1a.length(), topic2.length()-t1b.length()).indexOf('/') == -1)
         return true;
     }
+  }
+  else if ((i = topic1.indexOf('$')) >= 0)
+  {
+    // shared subscription for MQTTv5
+    if (topic1.startsWith("$share"))
+    {
+      return topic1.endsWith(topic2);
+    }
```

Comment:
I havnt read about shared subscriptions yet, only looked at the code.

Why is it not directly the second if statement? Why first the indexOf? According to the second if the $ is the first character anyway so startsWith should have better performance then too. Am I missing something here?

```suggestion
  else if (topic1.startsWith("$share")) // shared subscription for MQTTv5
  {
    return topic1.endsWith(topic2);
```

Reply:
Take a look a the man page here:  https://mosquitto.org/man/mosquitto-8.html.   Under "Broker Status", it is possible to have a topic starting with "$SYS".   

"Clients can find information about the broker by subscribing to topics in the $SYS hierarchy as follows."

So I only implemented "$share" at this point, as I selfishly only need the round-robin shared subscription for two esp32 to take turns in solving a task.

Reply:
`$SYS` is irrelevant here as its unrelated to shared topics?

As far as I understand the shared topics, the topic is always like this:
`$share/groupid/topic` (example: `$share/my-shared-subscriber-group/myhome/groundfloor/+/temperature`)

So shared topics always start with "$share". The first if is irrelevant as the second if is both more precise and better performance wise?

Also something I noticed: shared topic subscriptions also support wildcards (`+` and `#`) which this implementation currently isn't. For a first version not a problem but should be noted as a known issue I guess.

Shared topics look interesting, I should look into them more.